brave/brave-core
# Adds functionality to monthly contribution dropdown (Panel, #2)

Fixes: https://github.com/brave/brave-browser/issues/2245
Fixes: https://github.com/brave/brave-browser/issues/2823
UI: https://github.com/brave/brave-ui/pull/434

Preview:
<div style="width: 100%">
<div style="float: left; width: 50%">
<img width="442" alt="Screen Shot 2019-03-20 at 6 52 47 PM" src="https://user-images.githubusercontent.com/8732757/54729754-80a91480-4b42-11e9-953e-6b7a023c6085.png">
</div>
<div style="float: right; width: 50%">
<img width="483" alt="Screen Shot 2019-03-20 at 6 52 57 PM" src="https://user-images.githubusercontent.com/8732757/54729755-8272d800-4b42-11e9-810a-ef6367a670ef.png">
</div>
</div>

The brunt of the work for this was adding extension functions to support:

Adding a recurring contribution
Removing a recurring contribution
Retrieving recurring contributions

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Test Plan:
   1. Enable Brave Rewards from a clean profile, add tokens
   2. Navigate to a site such as `brave.com`
   3. Bring up the wallet panel.
   4. Ensure the monthly contribution dropdown is not shown
   5. Click "Send Tip" and make a monthly donation (mark "Make this monthly")
   6. Re-open the wallet panel
   7. Change monthly contribution amount to a different amount than the recurring tip
   8. Close and re-open panel
   9. Confirm that the value set in step 7 persists
   10. Confirm that the change is reflected in the donation table
   11. Confirm that setting a monthly amount of 0.0 removes the donation from the table.
 
   The above six steps should be tried in different ways with different values. Ex: Ensuring that the data persists across tabs with the same site open

## Reviewer Checklist:
